include/admin/entries.inc.php

```diff
@@ -340,8 +340,10 @@
             foreach ((array)$serendipity['GET']['sort'] as $k => $v) {
                 $qString .= '&amp;serendipity[sort]['. $k .']='. $v;
             }
-            foreach ((array)$serendipity['GET']['filter'] as $k => $v) {
-                $qString .= '&amp;serendipity[filter]['. $k .']='. $v;
+            if (isset($serendipity['GET']['filter'])) {
+                foreach ((array)$serendipity['GET']['filter'] as $k => $v) {
+                    $qString .= '&amp;serendipity[filter]['. $k .']='. $v;
```

Comment:
We need to check XSS injection here too, that looks a bit dangerous?

Reply:
I think these variables are filtered above, at https://github.com/s9y/Serendipity/blob/20c0bed069af1a2a7cc815b9a433ad9d637b798d/include/admin/entries.inc.php#L228. And they also support just a limited set of inputs. But I might miss something. It would be great if someone could double-check this.